lib/Threading/Pthreads.cpp

```diff
@@ -101,18 +101,23 @@ swift::threading_impl::thread_get_current_stack_bounds() {
 #if defined(__FreeBSD__)
   if (pthread_attr_init(&attr))
     return {};
-#endif
 
-  if (!pthread_getattr_np(pthread_self(), &attr)) {
-    if (!pthread_attr_getstack(&attr, &begin, &size)) {
-      stack_bounds result = { begin, (char *)begin + size };
-      pthread_attr_destroy(&attr);
-      return result;
-    }
+  if (pthread_attr_get_np(pthread_self(), &attr)) {
+    pthread_attr_destroy(&attr);
+    return {};
+  }
+#else
+  if (pthread_getattr_np(pthread_self(), &attr))
```

Comment:
What uses this codepath?  AFAICT, `pthread_getattr_np()` is a glibcism, but Linux and Android use the implementation in `Linux.cpp` instead.  Should I remove this (and clean up the rest)?

Reply:
I wouldn't be inclined to remove it, honestly. I don't know why I hadn't spotted that FreeBSD had `attr_get_np()` instead of `getattr_np()`; I wonder if the older code I was looking at had a `#define` in it somewhere to paper over the difference.

Linux folks don't _have to_ use the Linux threading package; they could choose to use the pthreads package or the C11 one if they wanted for some reason, so I think I'd keep the other code as a fallback.

Reply:
Sounds good to me.  Would you prefer I change the `#else` to an `#elif defined(__linux__)` and add an `#else #error` to catch others?

Reply:
I suppose you could change it to an `#elif defined(__linux__)` and then in the `#else` case just return `{}`, since we don't know how to get the stack bounds in that situation.

Reply:
Done, thanks.